scripts/mirror.js

```diff
@@ -0,0 +1,617 @@
+      );
+  }
+
+  return source;
+};
+
+/**
+ * @param {string|string[]|null} notes1
+ * @param {string|string[]|null} notes2
+ * @returns {string|string[]|null}
+ */
+const combineNotes = (notes1, notes2) => {
+  let newNotes = [];
+
+  if (notes1) {
+    if (typeof notes1 === 'string') {
+      newNotes.push(notes1);
+    } else {
+      newNotes.push(...notes1);
+    }
+  }
+
+  if (notes2) {
+    if (typeof notes2 === 'string') {
+      newNotes.push(notes2);
+    } else {
+      newNotes.push(...notes2);
+    }
+  }
+
+  newNotes = newNotes.filter((item, pos) => {
+    newNotes.indexOf(item) == pos;
+  });
+
+  if (newNotes.length == 0) {
+    return null;
+  }
+  if (newNotes.length == 1) {
+    return newNotes[0];
+  }
+
+  return newNotes;
+};
+
+/**
+ * @param {string|string[]|null} notes
+ * @param {RegExp} regex
+ * @param {string} replace
+ * @returns {string|string[]|null}
+ */
+const updateNotes = (notes, regex, replace) => {
+  if (notes === null || notes === undefined) {
+    return null;
+  }
+
+  if (typeof notes === 'string') {
+    return notes.replace(regex, replace);
+  }
+
+  let newNotes = [];
+  for (let note of notes) {
+    newNotes.push(note.replace(regex, replace));
+  }
+  return newNotes;
+};
+
+/**
+ * @param {SupportStatement} edgeData
+ * @param {SupportStatement} chromeData
+ * @returns {SupportStatement}
+ */
+const bumpVersionEdgeChromium = (edgeData, chromeData) => {
+  let newData = edgeData == undefined ? chromeData : edgeData;
+
+  let chromeFalse =
+    chromeData.version_added === false ||
+    chromeData.version_removed !== undefined;
+  let chromeNull = chromeData.version_added === null;
+
+  if (edgeData === undefined) {
+    newData.version_added = chromeFalse ? false : chromeNull ? null : '≤79';
+  } else {
+    if (!chromeFalse && !chromeNull) {
+      if (edgeData.version_added == true) {
+        newData.version_added = '≤18';
+      } else {
+        if (
+          chromeData.version_added == true ||
+          Number(chromeData.version_added) <= 79
+        ) {
+          if (edgeData.version_added == false) {
+            newData.version_added = '79';
+          } else if (edgeData.version_added == null) {
+            newData.version_added = '≤79';
+          }
+        } else {
+          newData.version_added == chromeData.version_added;
+        }
+      }
+    } else if (chromeFalse) {
+      if (edgeData.version_added && !edgeData.version_removed) {
+        newData.version_removed = '79';
+      }
+    }
+  }
+
+  let newNotes = combineNotes(
+    updateNotes(chromeData.notes, /Chrome/g, 'Edge'),
+    edgeData.notes,
+  );
+
+  if (newNotes) {
+    newData.notes = newNotes;
+  }
+
+  return newData;
+};
+
+/**
+ * @param {SupportStatement} data
+ * @param {string} destination
+ * @param {string} source
+ * @param {SupportStatement} originalData
+ */
+const bumpVersion = (data, destination, source, originalData) => {
```

Comment:
This function has a really high complexity and only two functions that are part of the core logic have been split out.
I wonder if instead we could split it up by the 8 destinations `bumpChromeAndroid`, `bumpEdge`, `bumpFirefoxAndroid`, `bumpOpera`, `bumpOperAndroid`, `bumpSafariiOS`, `bumpSamsungInternet`, `bumpWebViewAndroid`?
That way, when something is odd, it might be easier to get into the relevant function to debug things. And it would also be more readable and reviewable, I think. The two functions that have been split out now (`create_webview_range` and `bumpVersionEdgeChromium`) could then live inside the respective bump functions. What do you think about that?

`getMatchingBrowserVersion`, `getSource`, and note handling functions are fine as is. (Although I need to have a closer look at note handling given that we identified its logic as problematic).